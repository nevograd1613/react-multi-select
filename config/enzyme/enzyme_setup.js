// TODO: Remove this `raf` polyfill once the below issue is sorted
// https://github.com/facebookincubator/create-react-app/issues/3199#issuecomment-332842582
import raf from "./tempPolyfills";

import Enzyme from "enzyme";
import Adapter from "@cfaester/enzyme-adapter-react-18";

Enzyme.configure({ adapter: new Adapter() });
