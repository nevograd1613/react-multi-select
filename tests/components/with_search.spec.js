import React from "react";
import { mount } from "enzyme";
import ShallowRenderer from "react-shallow-renderer";
import withSearch from "../../src/components/with_search.js";

const CustomComponent = (props) => <div {...props} />;
const ListComponent = withSearch(CustomComponent);
const messages = { searchPlaceholder: "Search" };
const UserSearch = (props) => (
  <input type="text" placeholder={messages.searchPlaceholder} {...props} />
);

describe("With search", () => {
  test("Case default search component", () => {
    const renderer = new ShallowRenderer();
    const filterItems = jest.fn();
    const filteredItems = [{ id: 1, label: "item-1" }];

    const component = renderer.render(
      <ListComponent
        showSearch={true}
        searchValue="1"
        searchIcon={<img src="custom_icon" alt="" />}
        messages={messages}
        filterItems={filterItems}
        filteredItems={filteredItems}
      />,
    );
    expect(component).toMatchSnapshot();
  });
  test("Case custom search component", () => {
    const renderer = new ShallowRenderer();
    const filterItems = jest.fn();
    const filteredItems = [{ id: 11, label: "item-11" }];
    const component = renderer.render(
      <ListComponent
        showSearch={true}
        searchRenderer={UserSearch}
        searchIcon={<img src="custom_icon" alt="" />}
        searchValue="11"
        messages={messages}
        filteredItems={filteredItems}
        filterItems={filterItems}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  test("Case not search component", () => {
    const renderer = new ShallowRenderer();
    const filterItems = jest.fn();
    const component = renderer.render(
      <ListComponent
        showSearch={false}
        searchRenderer={UserSearch}
        searchIcon={<img src="custom_icon" alt="" />}
        searchValue="11"
        messages={messages}
        filterItems={filterItems}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  test("Case messages are passed ok", () => {
    const filterItems = jest.fn();
    const component = mount(
      <ListComponent
        showSearch={false}
        searchRenderer={UserSearch}
        searchIcon={<img src="custom_icon" alt="" />}
        searchValue="11"
        messages={messages}
        filterItems={filterItems}
      />,
    );
    const listComponent = component.find(ListComponent);
    expect(listComponent.props().messages).toEqual(messages);
  });
});
