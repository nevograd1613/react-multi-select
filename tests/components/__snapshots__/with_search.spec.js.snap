// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With search Case custom search component 1`] = `
<CustomComponent
  filteredItems={
    [
      {
        "id": 11,
        "label": "item-11",
      },
    ]
  }
  messages={
    {
      "searchPlaceholder": "Search",
    }
  }
>
  <UserSearch
    onChange={[MockFunction]}
    searchIcon={
      <img
        alt=""
        src="custom_icon"
      />
    }
    searchPlaceholder="Search"
    value="11"
  />
</CustomComponent>
`;

exports[`With search Case default search component 1`] = `
<CustomComponent
  filteredItems={
    [
      {
        "id": 1,
        "label": "item-1",
      },
    ]
  }
  messages={
    {
      "searchPlaceholder": "Search",
    }
  }
>
  <Search
    onChange={[MockFunction]}
    searchIcon={
      <img
        alt=""
        src="custom_icon"
      />
    }
    searchPlaceholder="Search"
    value="1"
  />
</CustomComponent>
`;

exports[`With search Case not search component 1`] = `
<CustomComponent
  messages={
    {
      "searchPlaceholder": "Search",
    }
  }
/>
`;
